clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1005,6 +1083,43 @@ public void testNoWakeupInCloseCommit() {
         assertFalse(capturedEvent.get().future().isCompletedExceptionally());
     }
 
+    @Test
+    public void testCloseAwaitPendingAsyncCommitIncomplete() {
+        time = new MockTime(1);
+        consumer = newConsumer();
+
+        // Commit async (incomplete)
+        doReturn(Fetch.empty()).when(fetchCollector).collectFetch(any(FetchBuffer.class));
+        doReturn(LeaderAndEpoch.noLeaderOrEpoch()).when(metadata).currentLeader(any());
+        final TopicPartition tp = new TopicPartition("foo", 0);
+        consumer.assign(Collections.singleton(tp));
+        consumer.seek(tp, 20);
+
+        consumer.commitAsync();
+        Exception e = assertThrows(KafkaException.class, () -> consumer.close(Duration.ofMillis(10)));
+        assertInstanceOf(TimeoutException.class, e.getCause());
```

Comment:
So this is where it may make sense to have special case to close and clear out the `consumer` instance, per the previous comment.

Reply:
True. In this case, we are actually testing `close`. But there are other test cases, which time-out similarly, but that's only an artifact and not the goal of the test.